### PR TITLE
[release/3][BACKPORT] fix: Bump prometheus image to v2.29.2

### PR DESCRIPTION
**Pull request by mesosphere-mergebot**
This is a backport of the following PR:

https://github.com/mesosphere/kubernetes-base-addons/pull/1302

**What type of PR is this?**

bug

**What this PR does/ why we need it**:

bump prometheus image to fix a cops ticket

**Which issue(s) this PR fixes**:

https://jira.d2iq.com/browse/D2IQ-78551
https://jira.d2iq.com/browse/COPS-7011

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix: Bump Prometheus to v2.29.2 which fixes head GC and pending readers race condition that caused issues with rule evaluations.
```

**Checklist**

* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.
Before merging, conflicts in the following files need to be resolved: 
addons/prometheus/prometheus.yaml

### CODE REVIEW

File: addons/prometheus/prometheus.yaml [hunk 1]
```diff
@@ -9,9 +9,13 @@ metadata:
     # on the cluster, this hack will trigger re-queue on Addons until one exists.
     kubeaddons.mesosphere.io/hack-requires-defaultstorageclass: "true"
   annotations:
+<<<<<<< HEAD
     catalog.kubeaddons.mesosphere.io/addon-revision: "0.47.0-3"
+=======
+    catalog.kubeaddons.mesosphere.io/addon-revision: "0.47.0-5"
+>>>>>>> 01aa629... fix: Bump prometheus image to v2.29.2
```
Comment (gracedo) [inline addons/prometheus/prometheus.yaml:16]:
```suggestion
    catalog.kubeaddons.mesosphere.io/addon-revision: "0.47.0-4"
```

**Review by dkoshkin — APPROVED** [review]

**Review by samvantran — APPROVED** [review]

**Comment by samvantran** [comment]
@gracedo do we worry about the dispatch test-* jobs or this is good to go?

**Comment by gracedo** [comment]
> @gracedo do we worry about the dispatch test-* jobs or this is good to go?

not sure :/ I vaguely recall these tests not ever passing for backports, but maybe @joejulian knows better

**Comment by gracedo** [comment]
Looks like `flagger` addon is failing to get deployed in both the failing tests 🤔

**Comment by joejulian** [comment]
This worked locally.

**Pull request merged**